### DISCUSSION TITLE
[Android] [Security] Pendo key restriction android?

### DISCUSSION THREAD

**Discussion by bugmagnet78** [pendo-io/pendo-mobile-sdk#136] [Q&A] [answered]
Hey Pendo Team,

We are currently using pendo for tracking user events in our android application, And we have some queries about the restrictions that can be done to the API key for android.

Can we restrict the API key to a single android application(like google maps) using the pendo dashboard, so that even if the API key is compromised the hacker cannot use the API key for anything else?

Or if the API Key is compromised, will it a big security threat, and what can an external user can do with our pendo API? 

Looking forward for a solution from you guys asap.

Thanks

**Comment by bugmagnet78** [comment]
@MikePendo @ypendo @pendo-github-admin @pendo-ops @AnatPendo

**Comment by DavidFaermanPendo** [comment, marked as answer]
Thanks for contacting us.
Will answer the second question first:

If the API Key is compromised, will it a big security threat, and what can an external user can do with our pendo API?

_If the Key is compromised, the attacker can not do anything. The key is only good for sending information to Pendo; it can NOT be used to retrieve any information._ 

Can we restrict the API key to a single android application(like google maps) using the pendo dashboard, so that even if the API key is compromised the hacker cannot use the API key for anything else?

_We do not currently have that feature. However, we do have the JWT signature to ensure you are the only one sending data. 
Please check this link [Send signed metadata with JWT](https://support.pendo.io/hc/en-us/articles/360039616892-Send-signed-metadata-with-JWT)_

**Reply by bugmagnet78** [reply]
Thank you @DavidFaermanPendo For a quick response!!!

Out of curiosity, what happen if another application also starts to send tracking events with the compromised Pendo API Key? Will it even show on the Pendo dashboard?

Thanks

**Reply by udilevin** [reply]
It will probably won't appear in the Pendo Dashboard (only in the activity timeline) since it will not 'match' any tagged pages or features. 
You can always exclude any visitor via the UI in order to remove their analytics from the Dashboard.

**Reply by bugmagnet78** [reply]
Thanks @udilevin for the response!!
